static/app/gettingStartedDocs/node/koa.tsx

```diff
@@ -0,0 +1,181 @@
+                    include: {
+                      user: false,
+                    },
+                  })
+                )
+              );
+
+              try {
+                await next();
+              } catch (err) {
+                reject(err);
+              }
+              resolve();
+            });
+          });
+        };
+
+        // this tracing middleware creates a transaction per request
+        const tracingMiddleWare = async (ctx, next) => {
+          const reqMethod = (ctx.method || "").toUpperCase();
+          const reqUrl = ctx.url && stripUrlQueryAndFragment(ctx.url);
+
+          // connect to trace of upstream app
+          let traceparentData;
+          if (ctx.request.get("sentry-trace")) {
+            traceparentData = Sentry.extractTraceparentData(
+              ctx.request.get("sentry-trace")
+            );
+          }
+
+          const transaction = Sentry.startTransaction({
+            name: \`\${reqMethod} \${reqUrl}\`,
+            op: "http.server",
+            ...traceparentData,
+          });
+
+          ctx.__sentry_transaction = transaction;
+
+          // We put the transaction on the scope so users can attach children to it
+          Sentry.getCurrentHub().configureScope((scope) => {
+            scope.setSpan(transaction);
+          });
+
+          ctx.res.on("finish", () => {
+            // Push \`transaction.finish\` to the next event loop so open spans have a chance to finish before the transaction closes
+            setImmediate(() => {
+              // if using koa router, a nicer way to capture transaction using the matched route
+              if (ctx._matchedRoute) {
+                const mountPath = ctx.mountPath || "";
+                transaction.setName(\`\${reqMethod} \${mountPath}\${ctx._matchedRoute}\`);
+              }
+              transaction.setHttpStatus(ctx.status);
+              transaction.finish();
+            });
+          });
+
+          await next();
+        };
+
+        app.use(requestHandler);
+        app.use(tracingMiddleWare);
+
+        // usual error handler
+        app.on("error", (err, ctx) => {
+          Sentry.withScope((scope) => {
+            scope.addEventProcessor((event) => {
+              return Sentry.addRequestDataToEvent(event, ctx.request);
+            });
+            Sentry.captureException(err);
+          });
+        });
+
+        app.listen(3000);
+        `,
+      },
+    ],
+  },
+  getUploadSourceMapsStep('https://docs.sentry.io/platforms/node/guides/koa/sourcemaps'),
+  {
+    type: StepType.VERIFY,
+    description: t(
+      "This snippet contains an intentional error and can be used as a test to make sure that everything's working as expected."
+    ),
+    configurations: [
+      {
+        language: 'javascript',
+        code: `
+        app.get("/debug-sentry", function mainHandler(req, res) {
+          throw new Error("My first Sentry error!");
+        });
```

Comment:
Koa doesn't have `app.get|post|put` etc. functions by default (TIL) but users need to additionally install the `@koa/router` package for them. Their [error handling example](https://github.com/koajs/examples/blob/master/errors/app.js) shows a basic test error in the middleware `use` function, so I suggest, we do this as well:

```suggestion
        app.use(async function () {
          throw new Error("My first Sentry error!");
        });
```